api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -2491,6 +2491,38 @@ message UpstreamInventoryHello {
   string ExternalUpgrader = 5;
   // ExternalUpgraderVersion identifies the external upgrader version. Empty if no upgrader is defined.
   string ExternalUpgraderVersion = 6;
+  // UpdaterV2Info is used by Teleport to send information about how the Teleport updater is doing.
+  UpdaterV2Info UpdaterV2Info = 7;
+}
+
+message UpdaterV2Info {
+  // UpdateGroup is the update group the agent's updater is part of.
+  // The autoupdate_agent_rollout resource controls when different groups update.
+  // Reporting the update group in the Hello allows us to track the update progress per group.
+  // See RFD 184 for more details.
+  string UpdateGroup = 1;
+  // UpdateUUID is the agent's updater UUID.
+  // Each updater has a UUID, this can be used in two cases:
+  // - To update individual agents when doing canary updates
+  // - To track the lowest non-updated agent to avoid a deadlock when doing a progressive rollout
+  bytes UpdateUUID = 2;
+  // UpdaterStatus represents any updater-related status information that the Teleport cluster
+  // should be aware of. For example, the fact the updater got disabled, the version pinned,
+  // or that the updater seems to not be running.
+  UpdaterStatus UpdaterStatus = 3;
+}
+
+enum UpdaterStatus {
+  // UpdateStatusUnknown means that we have no information about the updater status.
+  UpdaterStatusUnknown = 0;
```

Comment:
I wouldn't even glorify the enum value 0 with a comment nor assign it a specific meaning - if we need a dedicated "unknown" status, give it a nonzero variant, zero is for values that are missing altogether.

Reply:
Changed the zero value comment and name + made the enum linter-friendly.